beacon_node/beacon_chain/src/block_verification_types.rs

```diff
@@ -140,19 +146,29 @@ impl<E: EthSpec> RpcBlock<E> {
         Hash256,
         Arc<SignedBeaconBlock<E>>,
         Option<BlobSidecarList<E>>,
+        Option<DataColumnSidecarList<E>>,
     ) {
         let block_root = self.block_root();
         match self.block {
-            RpcBlockInner::Block(block) => (block_root, block, None),
-            RpcBlockInner::BlockAndBlobs(block, blobs) => (block_root, block, Some(blobs)),
+            RpcBlockInner::Block(block) => (block_root, block, None, None),
+            RpcBlockInner::BlockAndBlobs(block, blobs) => (block_root, block, Some(blobs), None),
+            RpcBlockInner::BlockAndDataColumns(block, data_columns) => {
+                (block_root, block, None, Some(data_columns))
+            }
         }
     }
     pub fn n_blobs(&self) -> usize {
         match &self.block {
-            RpcBlockInner::Block(_) => 0,
+            RpcBlockInner::Block(_) | RpcBlockInner::BlockAndDataColumns(_, _) => 0,
```

Comment:
I think match statements are executed in order from left to right. There might be a weird edge case here where both data columns and blobs are avail, but we return 0. If we're trying to support both blobs and data_columns initially, this edge case has the potential to cause issues

Reply:
Yep, you're right - perhaps we can add the `BlockAndBlobsAndDataColumns` variant to the other arm?

Reply:
So this type is used for RPC responses, so i don't think blobs and data column will coexist - we either request for `block + blobs` or `block + data columns` during lookups and sync, depending on the fork we're at.